src/bvals/boundary_conditions.cpp

```diff
@@ -134,66 +98,90 @@ void GenericBC(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
   const int offset = 2 * ref + (INNER ? -1 : 1);
 
   pmb->par_for_bndry(
-      label, nb, domain, coarse,
+      label, nb, domain, el, coarse,
       KOKKOS_LAMBDA(const int &l, const int &k, const int &j, const int &i) {
         if (!q.IsAllocated(l)) return;
         if (TYPE == BCType::Reflect) {
           const bool reflect = (q.VectorComponent(l) == DIR);
-          q(l, k, j, i) =
+          q(el, l, k, j, i) =
               (reflect ? -1.0 : 1.0) *
-              q(l, X3 ? offset - k : k, X2 ? offset - j : j, X1 ? offset - i : i);
+              q(el, l, X3 ? offset - k : k, X2 ? offset - j : j, X1 ? offset - i : i);
         } else {
-          q(l, k, j, i) = q(l, X3 ? ref : k, X2 ? ref : j, X1 ? ref : i);
+          q(el, l, k, j, i) = q(el, l, X3 ? ref : k, X2 ? ref : j, X1 ? ref : i);
         }
       });
 }
 
 void OutflowInnerX1(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
-  GenericBC<X1DIR, BCSide::Inner, BCType::Outflow>(rc, coarse);
+  using TE = TopologicalElement;
+  for (auto el : {TE::CC, TE::F1, TE::F2, TE::F3, TE::E1, TE::E2, TE::E3, TE::NN})
+    GenericBC<X1DIR, BCSide::Inner, BCType::Outflow>(rc, coarse, el);
 }
 
 void OutflowOuterX1(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
-  GenericBC<X1DIR, BCSide::Outer, BCType::Outflow>(rc, coarse);
+  using TE = TopologicalElement;
+  for (auto el : {TE::CC, TE::F1, TE::F2, TE::F3, TE::E1, TE::E2, TE::E3, TE::NN})
+    GenericBC<X1DIR, BCSide::Outer, BCType::Outflow>(rc, coarse, el);
 }
```

Comment:
Some point soon we really need to add meshdata versions of these.

Reply:
Agreed.